src/components/RichTextEditor/Plugins/LinkPlugin/LinkButton.tsx

```diff
@@ -3,19 +3,30 @@
 import { getHotkeyByPlatform } from '@components/RichTextEditor/helpers/getHotkeyByPlatform';
 import { getTooltip } from '@components/RichTextEditor/helpers/getTooltip';
 import { IconLink16 } from '@foundation/Icon/Generated';
-import { LinkToolbarButton } from '@udecode/plate';
+import { LinkToolbarButton, isRangeInSameBlock, useEventPlateId, usePlateEditorState } from '@udecode/plate';
 import React from 'react';
 import { ButtonWrapper, IconStylingWrapper, buttonClassNames, buttonStyles } from '../helper';
 import { PluginButtonProps } from '../types';
 
-export const LinkButton = ({ id }: PluginButtonProps) => (
-    <ButtonWrapper id={id}>
-        <LinkToolbarButton
-            tooltip={getTooltip(`Link\n${getHotkeyByPlatform('Ctrl+K')}`)}
-            icon={<IconStylingWrapper icon={<IconLink16 />} />}
-            classNames={buttonClassNames}
-            styles={buttonStyles}
-            actionHandler="onMouseDown"
-        />
-    </ButtonWrapper>
-);
+export const LinkButton = ({ id, editorId }: PluginButtonProps) => {
+    const editor = usePlateEditorState(useEventPlateId(editorId));
+    const isInSameBlock = isRangeInSameBlock(editor, {
+        at: editor.selection,
+    });
+
+    return (
+        <ButtonWrapper id={id}>
+            <LinkToolbarButton
+                tooltip={getTooltip(`Link\n${getHotkeyByPlatform('Ctrl+K')}`)}
+                icon={<IconStylingWrapper icon={<IconLink16 />} />}
+                classNames={{
+                    root: `${buttonClassNames.root} ${isInSameBlock ? '' : '!tw-cursor-not-allowed !tw-opacity-50'}`,
```

Comment:
maybe we could also add `!tw-cursor-not-allowed !tw-opacity-50` to the buttonClassNames such that we can reuse it also like buttonClassNames.disabled. wdyt?

Reply:
good point!